packages/extension/src/lib/extensionScripts.ts

```diff
@@ -77,21 +60,39 @@ export const useExtensionPermission = ({
       });
       client.setQueryData(contentScriptKey, true);
       await registerBrowserContentScripts();
-      window.open(companionPermissionGrantedLink, '_blank');
+
+      if (onPermission) {
+        await onPermission(true);
+      } else {
+        window.open(companionPermissionGrantedLink, '_blank');
+      }
+    } else {
+      trackEvent({
+        event_name: 'decline content scripts',
+        extra: JSON.stringify({ origin }),
+      });
+
+      if (onPermission) onPermission(false);
     }
 
     return granted;
   };
+};
 
-  return useMemo(
-    () => ({
-      contentScriptGranted,
-      requestContentScripts,
-      isFetched,
-      registerBrowserContentScripts,
-    }),
-    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [contentScriptGranted, isFetched],
+export const useContentScriptStatus = (): {
```

Comment:
Maybe just me missing a explanation, but why is this one still needed as duplicate?

Reply:
It's not actually a duplicate, we just export it as a separate function so we can attach it to the context the same way we do with `requestContentScripts` and the others